prometheus/promhttp/http.go

```diff
@@ -406,3 +435,46 @@ func httpError(rsp http.ResponseWriter, err error) {
 		http.StatusInternalServerError,
 	)
 }
+
+// NegotiateEncodingWriter reads the Accept-Encoding header from a request and
+// selects the right compression based on an allow-list of supported
+// compressions. It returns a writer implementing the compression and an the
+// correct value that the caller can set in the response header.
+func NegotiateEncodingWriter(r *http.Request, rw io.Writer, disableCompression bool, compressions []string) (_ io.Writer, encodingHeaderValue string, closeWriter func() error, _ error) {
+	w := rw
+
+	if disableCompression {
+		return w, string(Identity), nil, nil
+	}
+
+	// TODO(mrueg): Replace internal/github.com/gddo once https://github.com/golang/go/issues/19307 is implemented.
+	compression := httputil.NegotiateContentEncoding(r, compressions)
```

Comment:
```suggestion
	selected := httputil.NegotiateContentEncoding(r, compressions)
```

Perhaps more readable?